src/uu/head/src/app.rs

```diff
@@ -0,0 +1,254 @@
+                .overrides_with_all(&[constants::quiet_name(), constants::verbose_name()]),
+        )
+        .arg(
+            Arg::with_name(constants::zero_name())
+                .short("z")
+                .long("zero-terminated")
+                .help(constants::zero_help())
+                .overrides_with(constants::zero_name()),
+        )
+        .arg(Arg::with_name(constants::files_name()).multiple(true))
+}
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum Modes {
+    Lines(usize),
+    Bytes(usize),
+}
+
+pub fn parse_mode<F>(src: &str, closure: F) -> Result<(Modes, bool), String>
+where
+    F: FnOnce(usize) -> Modes,
+{
+    match parse::parse_num(src) {
+        Ok((n, last)) => Ok((closure(n), last)),
+        Err(reason) => match reason {
+            parse::ParseError::Syntax => Err(format!("'{}'", src)),
+            parse::ParseError::Overflow => {
+                Err(format!("'{}': Value too large for defined datatype", src))
+            }
+        },
+    }
+}
+
+//TODO: find better name for the `all_but_last` field
+//and apply it across the code
+#[derive(Debug)]
+pub struct HeadOptions {
+    pub quiet: bool,
+    pub verbose: bool,
+    pub zeroed: bool,
+    pub all_but_last: bool,
+    pub mode: Modes,
+    pub files: Vec<String>,
+}
+struct ArgIterator<IterType: Iterator<Item = OsString>> {
+    args: IterType,
+    flag: bool,
+    mode: Option<Modes>,
+}
+impl<IterType: Iterator<Item = OsString>> ArgIterator<IterType> {
+    fn new(args: IterType) -> ArgIterator<IterType> {
+        ArgIterator {
+            args,
+            flag: true,
+            mode: None,
+        }
+    }
+}
+
+impl<IterType: Iterator<Item = OsString>> Iterator for ArgIterator<IterType> {
+    type Item = OsString;
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.flag {
+            if self.mode.is_none() {
+                match self.args.next() {
+                    Some(oss) => {
+                        if oss == "head" {
+                            return Some(oss);
+                        }
+                        let arg_str = &oss.clone().into_string().unwrap_or_else(|_|"".to_owned());
+                        if let Some(res) = parse::parse_obsolete(&arg_str) {
+                            match res {
+                                Ok((n, b)) => {
+                                    if b {
+                                        self.mode = Some(Modes::Bytes(n));
+                                        Some(OsString::from("-c"))
+                                    } else {
+                                        self.mode = Some(Modes::Lines(n));
+                                        Some(OsString::from("-n"))
+                                    }
+                                }
+                                Err(e) => match e {
+                                    parse::ParseError::Overflow => {
+                                        eprintln!(
```

Comment:
crash! macro used widely for this operation in codebase, it 'd be better to use it for consistency.